### PR TITLE
add more valid test cases we support as part of smoke

### PR DESCRIPTION
**Pull request by amarts**
Smoke tests currently use 'replica 2' which we say we are discouraging. Need to have valid options used for smoke, so we have confidence in the volume type.

@ShyamsundarR @vbellur @nigelbabu @raghavendra-talur

### CODE REVIEW

**Comment by nigelbabu** [comment]
I'll test this and move it to production.

File: smoke.sh [hunk 1]
```diff
@@ -21,13 +21,26 @@ function cleanup()
...
 
-function start_fs()
+function start_fs_with_arbiter()
 {
     mkdir -p $P/export;
     chmod 0755 $P/export;
 
     glusterd;
-    gluster --mode=script volume create $V replica 2 $H:$P/export/export{1,2,3,4} force;
+    gluster --mode=script volume create $V replica 3 arbiter 1 $H:$P/export/export{1,2,3,4,5,6} force;
```
Comment (raghavendra-talur) [inline smoke.sh]: Any particular reason to choose arbiter over plain replica 3?

**Comment by raghavendra-talur** [comment]
Apart from one minor comment, everything else looks good.

File: smoke.sh (hunk 1 above)
Comment (amarts) [reply smoke.sh]: I thought adding arbiter here will have more lines covered than plain replica 3.

**Comment by nigelbabu** [comment]
I'm going to defer to you all on this one. Please approve + merge and ping me so I can deploy it.

**Comment by ShyamsundarR** [comment]
The code change is fine, but I am not sure we are capturing failures correctly.

In the past there was a hang situation and WATCHDOG killed this and smoke failed (at least this part of smoke). In finish() we are just checking last RET, and that could be anything. So if arbiter test fails, we would just continue and run disperse, and the failure is really trapping only the last CLI failure, this seems wrong (even from before).

Am I missing something?

File: smoke.sh (hunk 1 above)
Comment (raghavendra-talur) [reply smoke.sh]: Ok, makes sense.

**Comment by amarts** [comment]
@ShyamsundarR you are right! Let me see how to fix this, and update the PR.

**Comment by amarts** [comment]
@ShyamsundarR this change is good enough! `trap finish EXIT` captures any exit status other than 0, and hence will trigger failure if it actually fails.

IMO, good to go.

**Comment by ShyamsundarR** [comment]
> @ShyamsundarR this change is good enough! trap finish EXIT captures any exit status other than 0, and hence will trigger failure if it actually fails.

@amarts this is true for the last command that executed, right? Also, I am not stating this patch introduces any issue, just that we are not looking for failures from before as well.

Here is what I see,
  - Assume line 33 failed to mount the volume, the RET is not captured anywhere, and a exit is not triggered (which would enable the trap to execute and capture RET and fail the test)
  - If mount fails, dbench will still run (on the local directory), so will posix_compliance.sh, and again we will not capture any errors

So, ideally, each command should be checked for a *sane* return and we should exit from there, for the trap to catch it and fail the test.

Like stated this is an existing problem IMO, so we can patch it up next, rather than overloading this patch.

**Comment by ShyamsundarR** [comment]
> So, ideally, each command should be checked for a sane return and we should exit from there, for the trap to catch it and fail the test.

So the set -e at the top does this for us, any non-zero return from any command will trigger an exit and hence the trap and other things. @nigelbabu helped clarify this.

So as the new commands used are the same as the old ones, the assumption that we are not dealing with any non-zero returns from any command, we can ignore the above comment.

**Pull request merged**